File: app/src/main/java/de/sneak/sneakpeek/adapter/PreviousMoviesAdapter.java
package de.sneak.sneakpeek.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import de.sneak.sneakpeek.R;

public class PreviousMoviesAdapter extends RecyclerView.Adapter<PreviousMoviesAdapter.MovieViewHolder> {

    private List<String> movieTitles;

    public PreviousMoviesAdapter(List<String> movieTitles) {
        this.movieTitles = movieTitles;
    }

    public void addAll(List<String> movieTitles) {
        this.movieTitles.clear();
        for (String title : movieTitles) {
            this.movieTitles.add(title);
            notifyItemInserted(movieTitles.indexOf(title));
        }
    }

    @Override
    public MovieViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.layout_movie_list_element, parent, false);

        return new MovieViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(MovieViewHolder holder, int position) {
        holder.title.setText(movieTitles.get(position));
    }

    @Override
    public int getItemCount() {
        return movieTitles.size();
    }

    public static class MovieViewHolder extends RecyclerView.ViewHolder {

        public TextView title;

        public MovieViewHolder(View itemView) {
            super(itemView);
            title = (TextView) itemView.findViewById(R.id.movie_list_item_title);
        }
    }
}

File: app/src/main/java/de/sneakpeek/adapter/PreviousMoviesAdapter.kt
package de.sneakpeek.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import de.sneakpeek.R


class PreviousMoviesAdapter(private val movieTitles: MutableList<String>, private val clickListener: PreviousMovieViewHolder.ClickListener) :
        RecyclerView.Adapter<PreviousMoviesAdapter.PreviousMovieViewHolder>() {

    fun addAll(movieTitles: List<String>) {
        this.movieTitles.clear()
        this.movieTitles.addAll(movieTitles)
        notifyDataSetChanged()
    }

    fun getTitle(position: Int): String {
        return movieTitles.get(position)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PreviousMovieViewHolder {
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.layout_previous_movie_list_element, parent, false)

        return PreviousMovieViewHolder(itemView, clickListener)
    }

    override fun onBindViewHolder(holder: PreviousMovieViewHolder, position: Int) {
        holder.date.text = "" + (position + 1) + "."
        holder.title.text = movieTitles[position]
    }

    override fun getItemCount(): Int {
        return movieTitles.size
    }

    class PreviousMovieViewHolder internal constructor(itemView: View, private val clickListener: PreviousMovieViewHolder.ClickListener?) :
            RecyclerView.ViewHolder(itemView), View.OnClickListener {

        internal var title: TextView = itemView.findViewById(R.id.previous_movie_title) as TextView
        internal var date: TextView = itemView.findViewById(R.id.previous_movie_date) as TextView

        init {
            itemView.setOnClickListener(this)
        }

        override fun onClick(v: View) {
            clickListener?.onItemClicked(adapterPosition)
        }

        interface ClickListener {
            fun onItemClicked(position: Int)
        }
    }
}
